geth/rpc/call_raw.go

```diff
@@ -175,7 +175,11 @@ func newSuccessResponse(result json.RawMessage, id json.RawMessage) string {
 		},
 		Result: result,
 	}
-	data, _ := json.Marshal(msg)
+	data, err := json.Marshal(msg)
+	if err != nil {
+		return newErrorResponse(errInvalidMessageCode, err, id)
+	}
+
```

Comment:
It is really unlikely to have error here with current code: as far as I know Marshal can fail only on unsupported types (like marshalling a function). But yeah, it won't hurt.